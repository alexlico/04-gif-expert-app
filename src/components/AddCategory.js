import React, { useState } from 'react'
import PropTypes from 'prop-types'
export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('')
    
    const handlerInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories((cat) => [inputValue, ...cat])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlerSubmit}>
            <input 
                type="text" 
                value={inputValue}
                onChange={handlerInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}