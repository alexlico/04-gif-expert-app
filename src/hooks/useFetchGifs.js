import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    })
    useEffect(() => {
        setTimeout(() => {
            getGifs(category).then((images) => {
                setState({
                    data: images,
                    loading: false
                })
            })
        }, 2000);
    
    }, [category])

    return state;
}
