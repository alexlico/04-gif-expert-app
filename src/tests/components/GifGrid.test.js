import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre el componente <GifGrid/>', () => {
    const category = 'Goku';

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe mostrar los elementos de imagenes cuando se cargan del useFetchGifs', () => {
        const images = [{
            id: 86876354,
            url: 'https://localhost/image.png',
            title: 'Esto es un titulo'
        }];
        useFetchGifs.mockReturnValue({
            data: images,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(images.length)
    
    })
})
