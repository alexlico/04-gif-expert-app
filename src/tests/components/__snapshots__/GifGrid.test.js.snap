// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre el componente <GifGrid/> Debe mostrar los elementos de imagenes cuando se cargan del useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={86876354}
      key="86876354"
      title="Esto es un titulo"
      url="https://localhost/image.png"
    />
  </div>
</Fragment>
`;
